test/utils.jl

```diff
@@ -273,6 +276,50 @@ function test_manifold(M::Manifold, pts::AbstractVector;
         @test isapprox(M, pts[1], vector_transport_to(M, pts[1], v1, pts[1]), v1)
     end
 
+    for btype ∈ basis_types_vecs
+        x = pts[1]
+        b = basis(M, x, btype)
+        @test isa(b, AbstractPrecomputedOrthonormalBasis)
+        N = length(b.vectors)
+        @test N == manifold_dimension(M)
+
+        # test orthonormality
+        for i in 1:N
+            @test norm(M, x, b.vectors[i]) ≈ 1
+            for j in i+1:N
+                @test real(inner(M, x, b.vectors[i], b.vectors[j])) ≈ 0 atol = sqrt(eps(eltype(x)))
+            end
+        end
+        if isa(btype, ProjectedOrthonormalBasis)
+            # check projection idempotency
+            for i in 1:N
+                @test project_tangent(M, x, b.vectors[i]) ≈ b.vectors[i]
+            end
+        end
+
+        if !isa(btype, ProjectedOrthonormalBasis)
+            v1 = inverse_retract(M, x, pts[2], default_inverse_retraction_method)
+            vb = represent_in_basis(M, x, v1, btype)
+
+            @test represent_in_basis(M, x, v1, b) ≈ represent_in_basis(M, x, v1, btype)
+            @test inverse_represent_in_basis(M, x, vb, b) ≈ inverse_represent_in_basis(M, x, vb, btype)
+        end
+    end
+
+    for btype ∈ (basis_types_to_from..., basis_types_vecs...)
```

Comment:
Should also check for orthonormality here for `AbstractOrthonormalBasis`.

Reply:
Probably by using `inverse_represent_in_basis` on (1, 0, ..., 0), (0, 1, 0, ..., 0), ..., (0, 0, ..., 0, 1) and checking returned vectors, right?